src/main/webapp/app/shared/category-selector/category-selector.component.ts

```diff
@@ -52,18 +50,18 @@ export class CategorySelectorComponent implements OnInit, OnChanges {
         );
     }
 
-    private categoriesAsStringArray() {
+    private categoriesAsStringArray(): string[] {
         if (!this.categories) {
             return [];
         }
-        return this.categories.map((exerciseCategory) => exerciseCategory.category?.toLowerCase());
+        return this.categories.map((exerciseCategory) => exerciseCategory.category!.toLowerCase());
     }
 
-    private existingCategoriesAsStringArray() {
+    private existingCategoriesAsStringArray(): string[] {
         if (!this.existingCategories) {
             return [];
         }
-        return this.existingCategories.map((exerciseCategory) => exerciseCategory.category?.toLowerCase());
+        return this.existingCategories.map((exerciseCategory) => exerciseCategory.category!.toLowerCase());
     }
 
     // if the user types in something, we need to filter for the matching categories
```

Comment:
According to the coverage report, this method is not tested at all. Could you add tests that include the user input and the resulting filter?

Reply:
@julian-christl Good catch, I added a test for this case now 👍🏼 
Also, I updated the percentages (They're in the order as described in the PR template statements | branches )